### PR TITLE
DEV-1997: Implement sftp backup option.

### PR DESCRIPTION
**Pull request by StavoveiC**

### CODE REVIEW

File: platform_db_backup.sh [hunk 1]
```diff
@@ -1,72 +1,69 @@
...
+  if [[ $SITE == '[^' ]]; then
+    continue;
+  fi
 
-  done
-
-  # loop through backups and upload to S3 until none left.
-  find $HOME/drush-backups -type f -name *gz -print | while read dump;
-  do
-    upload_dump_to_s3 $dump;
```
Comment (petarbasic) [inline platform_db_backup.sh:67]: Was this part removed on purpose? 
After these changes it seems to me that only files are being backed up, not the DB

File: platform_db_backup.sh [hunk 2]
```diff
@@ -1,72 +1,69 @@
...
+      tar -zcvf drush-backups/${PLATFORM_APPLICATION_NAME}/files-${SITE}/${PLATFORM_BRANCH}/files/${DAY}/files-${DAY}.tar.gz  docroot/sites/${SITE}/files
+    fi
+  fi
+done
 
-  # clean up remaining files
-  find $HOME/drush-backups -mindepth 1 -type d -empty -delete
-fi
+# copy files from newly created directory to SFTP.
+scp -i .ssh/id_rsa.pub -P $SFTP_PORT -r ./drush-backups/${PLATFORM_APPLICATION_NAME} ${SFTP_USERNAME}@${SFTP_SERVER}:~/${SFTP_DIRECTORY}
```
Comment (petarbasic) [inline platform_db_backup.sh]: It looks like this will run regardless of the chosen backup server (it will run always). What if only aws is wanted?

Comment (petarbasic) [inline platform_db_backup.sh]: I guess this adds new files to the directory without overwriting the whole dir, right?

File: platform_db_backup.sh (hunk 1 above)
Comment (StavoveiC) [reply platform_db_backup.sh:67]: no, it was not on purpose. fixed it now

File: platform_db_backup.sh (hunk 2 above)
Comment (ivangrozni) [reply platform_db_backup.sh]: it is much better to use `rsync` than `scp`. `rsync` will not error out without any warning. afaik `scp` doesn't have any mechanism to check whether files were really transferred.

**Review by ivangrozni — APPROVED** [review]

**Pull request merged**